src/BenchmarkDotNet/ConsoleArguments/ConfigParser.cs

```diff
@@ -356,6 +358,22 @@ private static Job CreateJobForGivenRuntime(Job baseJob, string runtimeId, Comma
                     builder.TargetFrameworkMoniker(runtime.MsBuildMoniker);
 
                     return baseJob.WithRuntime(runtime).WithToolchain(builder.ToToolchain());
+                case RuntimeMoniker.Wasm:
+                    var wasmRuntime = runtimeMoniker.GetRuntime();
+
+                    WasmSettings wasmSettings = new WasmSettings(wasmMainJS: options.WasmMainJS,
+                                                                 wasmJavaScriptEngine: options.WasmJavascriptEnginePath,
+                                                                 wasmjavaScriptEngineArguments: options.WasmJavaScriptEngineArguments
+                                                                 );
+
+                    IToolchain toolChain = new WasmToolChain(name: "Wasm",
+                                                             targetFrameworkMoniker: wasmRuntime.MsBuildMoniker,
+                                                             cliPath: options.CliPath.FullName,
+                                                             packagesPath: options.RestorePath?.FullName,
+                                                             wasmSettings: wasmSettings,
+                                                             timeout: timeOut ?? NetCoreAppSettings.DefaultBuildTimeout);
+
+                        return baseJob.WithRuntime(runtimeMoniker.GetRuntime()).WithToolchain(toolChain);
```

Comment:
we already have the runtime in `wasmRuntime` local variable:

```suggestion
                        return baseJob.WithRuntime(wasmRuntime).WithToolchain(toolChain);
```